internal/docstore/mongodocstore/mongo.go

```diff
@@ -649,22 +652,31 @@ func (c *collection) unorderedBulkWrite(ctx context.Context, iactions []indexedA
 			if newID != nil {
 				newIDs[i] = newID
 			}
+			if err == nil {
```

Comment:
I don't think we need this, because I think a create will return an error if the key already exists.